source/errors.ts

```diff
@@ -1,11 +1,16 @@
-'use strict';
-const urlLib = require('url');
-const http = require('http');
-const PCancelable = require('p-cancelable');
-const is = require('@sindresorhus/is');
-
-class GotError extends Error {
-	constructor(message, error, options) {
+import urlLib from 'url';
+import http, {IncomingHttpHeaders} from 'http';
+import PCancelable from 'p-cancelable';
+import is from '@sindresorhus/is';
+import {Response, Timings, Options} from './utils/types';
+import {TimeoutError as TimedOutError} from './utils/timed-out';
+
+type ErrorWithCode = (Error & { code?: string }) | { code?: string };
+
+export class GotError extends Error {
+	code?: string;
+
+	constructor(message: string, error: ErrorWithCode, options: Options) {
```

Comment:
Why not make error optional and use `undefined` instead `{}`?

Reply:
Since the `options` argument is also required, we would need to do `Error | undefined` since we can't do `code?: Error, options: Options`.

Reply:
Seems good. Let's go with that.